engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
```

Comment:
I apologize as I haven't read anything but @Horneth 's last comment so I might be way off base or talking about something different altogether.

The idea is that there's one of these, IMO it should hang directly off of the system's kernel. in the future it might be a simple router w/ stuff behind it but that address would be a singular address that anything needing validation would send stuff to